eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/sorted/MutableSortedBag.java

```diff
@@ -271,4 +272,7 @@
 
     @Override
     MutableSortedBag<T> drop(int count);
+
+    @Override
+    MutableSet<T> asSet();
```

Comment:
I tried but the main problem is coming from `MutableSortedMap.keySet` (the underlying map in `TreeBag`) which returns only a `MutableSet` with the following comment. So if something can be done about that, then yes this one could be sorted IMHO.
```
    /**
     * The underlying set for the keys is sorted in ascending order according to their natural ordering or a custom comparator.
     * However, Java 5 TreeMap returns a keySet that does not inherit from SortedSet therefore we have decided to
     * return the keySet simply as a MutableSet to maintain Java 5 compatibility.
     */
    //todo: Change return type to MutableSortedSet when we move to Java 6
```

Reply:
I've pushed some changes where I simply cast the keySet to bypass this, should I open an issue to solve the todo above?

Reply:
Actually I can't because `MutableSortedSet` already wrapped it in `SetAdapter` unfortunately...

Reply:
And we can't simply wrap the keySet instead in SortedSetAdapter, because `MutableSortedSet` is not a subclass of `MutableSet` but of `MutableSetIterable`